Terminal file links do not recognize the line number

Hi, similar to this issue https://github.com/microsoft/vscode/issues/171880 but in my case the line numbers are within parenthesis and are not recognized.

![MTL-BJ465](https://github.com/microsoft/vscode/assets/1246341/d811e5b9-92a1-4424-8441-ecc55776aa4f)



**Current behavior:** The file is recognized and I can click on it to open it but will not consider the linenumber
**Expected behavior:** When the file is opened after click, it takes me to the right line number



<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- 🕮 Read our guide about submitting issues: https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions -->
<!-- 🔎 Search existing issues to avoid creating duplicates. -->
<!-- 🧪 Test using the latest Insiders build to see if your issue has already been fixed: https://code.visualstudio.com/insiders/ -->
<!-- 💡 Instead of creating your report here, use 'Report Issue' from the 'Help' menu in VS Code to pre-fill useful information. -->
<!-- 🔧 Launch with `code --disable-extensions` to check. -->
Does this issue occur when all extensions are disabled?: Yes

<!-- 🪓 If you answered No above, use 'Help: Start Extension Bisect' from Command Palette to try to identify the cause. -->
<!-- 📣 Issues caused by an extension need to be reported directly to the extension publisher. The 'Help > Report Issue' dialog can assist with this. -->
- **VS Code Version: 1.82.2 <-- LATEST**
- OS Version: Win 10

Steps to Reproduce:

1. Have a cpp file with some syntax error
2. Compile that cpp file with FastBuild


Note: reopening this issue https://github.com/microsoft/vscode/issues/192457 after updating to latest version and verifying the bug is still there.

I think I can reproduce:

![image](https://github.com/microsoft/vscode/assets/2193314/92093254-d5fa-437c-9bf6-e91eeb169f09)

This is only on the prompt line of `echo "C:\Github\microsoft\vscode\README.md(10): error C2891: 'Blah': Illegal blah"`, the line with (5) works fine.

It's weird that it happens sometimes but not others.

Ok got the repro:

- Open pwsh
- Run `echo C:\Github\microsoft\vscode\README.md` (or some other file that exists)
- Run `PS C:\Github\microsoft\vscode> echo "C:\Github\microsoft\vscode\README.md(10): error C2891: 'Blah': Illegal blah"`

This makes the upwards scan succeed as it found `C:\Github\microsoft\vscode\README.md` which matches eslint's link format.

--- a/src/vs/workbench/contrib/terminalContrib/links/browser/terminalMultiLineLinkDetector.ts
+++ b/src/vs/workbench/contrib/terminalContrib/links/browser/terminalMultiLineLinkDetector.ts
@@ -33,7 +33,7 @@ const lineNumberPrefixMatchers = [
 	// Eslint:
 	//   /some/file
 	//     16:5  error ...
-	/ *(?<link>(?<line>\d+):(?<col>\d+)?)/
+	/^ *(?<link>(?<line>\d+):(?<col>\d+)?)/
 ];
 
 const gitDiffMatchers = [

